lib/src/components/stepper/stepper.dart

```diff
@@ -0,0 +1,463 @@
+                        ),
+                      ),
+                      _getHeaderText(index),
+                    ],
+                  ),
+                ),
+                if (!_isLast(index))
+                  Expanded(
+                    child: Container(
+                      key: Key('line$index'),
+                      margin: const EdgeInsets.only(
+                        top: ZetaSpacing.x7,
+                        right: ZetaSpacing.b,
+                        left: ZetaSpacing.b,
+                      ),
+                      height: ZetaSpacing.x0_5,
+                      decoration: BoxDecoration(
+                        borderRadius: ZetaRadius.full,
+                        color: switch (widget.steps[index].type) {
+                          ZetaStepType.complete => _colors.green.shade50,
+                          ZetaStepType.disabled => _colors.borderSubtle,
+                          ZetaStepType.enabled => _colors.blue.shade50,
+                        },
+                      ),
+                    ),
+                  ),
+              ],
+            ];
+
+            final List<Widget> stepPanels = <Widget>[];
+            for (int i = 0; i < widget.steps.length; i += 1) {
+              stepPanels.add(
+                Visibility(
+                  maintainState: true,
+                  visible: i == widget.currentStep,
+                  child: widget.steps[i].content,
+                ),
+              );
+            }
+
+            return Column(
+              children: [
+                Material(
+                  color: Colors.transparent,
+                  child: Container(
+                    margin:
+                        const EdgeInsets.symmetric(horizontal: ZetaSpacing.m),
+                    child: Row(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: children,
+                    ),
+                  ),
+                ),
+                Expanded(
+                  child: ListView(
+                    controller: widget.controller,
+                    physics: widget.physics,
+                    children: <Widget>[
+                      AnimatedSize(
+                        curve: Curves.fastOutSlowIn,
+                        duration: kThemeAnimationDuration,
+                        child: Column(
+                          crossAxisAlignment: CrossAxisAlignment.stretch,
+                          children: stepPanels,
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            );
+          },
+        ),
+    };
+  }
+}
+
+/// Zeta step used in [ZetaStepper]. The step can have a title and subtitle,
+/// an icon within its circle, some content and a state that governs its
+/// styling.
+class ZetaStep {
+  /// Creates a step for a [ZetaStepper].
+  const ZetaStep({
+    required this.title,
+    required this.content,
```

Comment:
I think content should be optional as this component could be used statically without user interaction to display progress